### PR TITLE
Correct Date errors, Add more customization

### PR DESCRIPTION
**Pull request by mshadbolt**
## `src/csv_to_ingest.py`
* Added ability to specify a `--prefix` that is prepended to all identifiers in the dataset
* Added ability to specify `--sample` number of donors, equally spread across the programs in large_dataset
* Added ability to specify `--donors-per-program` and `--number-of-programs` for finer control of how the subsampling is done

## `requirements.txt`
* update to latest clinical_etl for the extra date validation

## `post_processing.py`
* Shifting dates to ensure that:
  * `date_of_death.month_interval` less than 10 shifted to 20 to give some buffer for the below calculations
  * The first PrimaryDiagnosis has an date interval of 0
  * `date_of_diagnosis` is before `date_of_death`
  * `treatment_start_date` is after `date_of_diagnosis`
  * `treatment_start_date` is before `date_of_death`
  * `treatment_end_date` is after `treatment_start_date`
  * `treatment_end_date` is before `date_of_death`

## small/medium/large csv files
* Updated by rerunning `json_to_csv.py` based on the latest edits to `post_processing.py` with date correction for date of death, treatment dates

## `README.md`
* Add instructions for using the new customization options

## `.gitignore`
Add more output files to the gitignore

## To test:

- Try out running the `src/csv_to_ingest.py` script with any args that you like
- Try ingesting into local or dev

### CODE REVIEW

File: src/csv_to_ingest.py [hunk 1]
```diff
@@ -14,19 +54,129 @@ def parse_args():
...
+    else:
+        size = size_mapping[args.size]
+        manifest_path = f"{repo_dir}/{size}_dataset_csv/"
+        dataset_path = f"{manifest_path}raw_data"
+        if args.prefix:
+            replace_identifiers(args.prefix, dataset_path)
+            dataset_path = f"{repo_dir}/custom_dataset_csv-{args.prefix}/raw_data"
+
+    packets, errors = CSVConvert.csv_convert(input_path=dataset_path, manifest_file=f"{manifest_path}/manifest.yml",
+                                             minify=True, index_output=False)
```
Comment (lilyyangyi301) [inline src/csv_to_ingest.py:214]: Got an error when running `python src/csv_to_ingest.py --size m`:
Screenshot 2024-04-08 at 2 28 06 PM

**Review by lilyyangyi301 — CHANGES_REQUESTED** [review]

File: src/csv_to_ingest.py (hunk 1 above)
Comment (mshadbolt) [reply src/csv_to_ingest.py:214]: Can you try uninstalling clinical_etl then reinstalling with the latest requirements? I think maybe it is using an older version that doesn't have the minify argument
```
pip uninstall clinical_etl
pip install -r requirements.txt
```
It should install v2.2.0 of clinical_etl

**Review by lilyyangyi301 — APPROVED** [review]
LGTM!

**Review by daisieh — APPROVED** [review]
As far as I can tell, this works as I'd expect. Should I be able to generate datasets that have no validation warnings at all, or is that out of scope? The datasets that I generated did successfully ingest.

**Comment by mshadbolt** [comment]
Thanks for testing it out. I think I consider fixing all the warnings out of scope for now. Perhaps it is a good idea to have at least one dataset or program with no warnings though

**Pull request merged**